scripts/upload_demo_to_space.py

```diff
@@ -1,78 +1,159 @@
 import argparse
 import pathlib
 import shutil
+import sys
 import tempfile
 import textwrap
-from typing import Optional
+from typing import List, Optional
 
-import huggingface_hub
+from huggingface_hub import CommitOperationAdd, HfApi
 
 
 def upload_demo_to_space(
     demo_name: str, space_id: str, hf_token: str, gradio_version: Optional[str]
 ):
-    """Upload a demo in the demo directory to a huggingface space.
+    """
+    Upload a demo from the demo directory to a Hugging Face Space in chunks of 50 files per commit.
+
     Args:
-        demo_name: The name of the demo to upload.
-        space_id: The id of the space to upload the demo to.
-        hf_token: HF api token. Need to have permission to write to space_id for this to work.
-        gradio_version: If not None, will set the gradio version in the created space to the given version.
+        demo_name (str): The name of the demo to upload.
+        space_id (str): The ID of the space to upload the demo to (e.g., username/space_name).
+        hf_token (str): Hugging Face API token with write permissions to the space.
+        gradio_version (Optional[str]): If provided, sets the Gradio version in the created space.
+
+    Returns:
+        str: URL of the uploaded Hugging Face Space.
     """
 
+    print(f"Uploading demo '{demo_name}' to space '{space_id}'...")
+
+    def split_into_chunks(lst: List, n: int) -> List[List]:
+        """
+        Splits a list into chunks of size n.
+
+        Args:
+            lst (List): The list to split.
+            n (int): The size of each chunk.
+
+        Returns:
+            List[List]: A list of chunks.
```

Comment:
nit: unnecessary docstring as what this method does is quite clear from the name and code